haicrypt/hcrypt_ctx_tx.c

```diff
@@ -74,9 +74,10 @@ int hcryptCtx_Tx_Rekey(hcrypt_Session *crypto, hcrypt_Ctx *ctx)
 	HCRYPT_PRINTKEY(ctx->sek, ctx->sek_len, "sek");
 
 	/* Regenerate KEK if Password-based (uses newly generated salt and sek_len) */
-	if ((0 < ctx->cfg.pwd_len)
-	&&	(0 > (iret = hcryptCtx_GenSecret(crypto, ctx)))) {
-		return(iret);
+	if (0 < ctx->cfg.pwd_len) {
```

Comment:
As you already fixed the Yoda Conditions in `iret` comparison, you could fix it here, too. [suggestion]